src/tests/multihost/alltests/test_rfc2307.py

```diff
@@ -106,3 +106,87 @@ def test_0001_bz1362023(self, multihost, backupsssdconf):
         user = f'\\ foo1@{domain_name}'
         cmd = multihost.client[0].run_command(f'id {user}', raiseonerr=False)
         assert cmd.returncode != 0
+
+    @staticmethod
+    def test_0002_gecos_finger(multihost):
+        """
+        :title: IDM-SSSD-TC: rfc2307: user with root user membership
+        :id: aaafdfd6-02cd-430d-a1ef-3291a9f54848
+        :description: a user, with one of it's group having gid zero,
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=748857
+        :steps:
+        1. Create user and two groups with a different ids and one group with
+          with zero gid
+        2. Restart SSSD with cleared cache
+        3. Fetch user information using 'id'
+        4. Confirm user is able to log in via ssh
+        :expectedresults:
+        1. Should succeed
+        2. Should succeed
+        3. Should succeed
+        4. Should succeed
+        """
+        client = sssdTools(multihost.client[0])
+        domain_name = client.get_domain_section_name()
+        r_num = random.randint(9999, 99999)
+        usr = f'user{r_num}'
+        usr_info = {'cn': usr, 'uid': usr,
+                    'uidNumber': f'{r_num}',
+                    'gidNumber': f'{r_num}'}
+        usr_grp(multihost, usr_info, 'user')
+        user = f'{usr}@{domain_name}'
+        memberdn = f'uid={usr},ou=People,{ds_suffix}'
+        grp_dict = {f'grp{r_num}': r_num,
+                    f'grp{r_num+1}': r_num+1,
+                    f'root{r_num}': '0'}
+        for obj in grp_dict:
+            group_info = {'cn': obj,
+                          'gidNumber': f'{grp_dict.get(obj)}',
+                          'memberUid': f'{usr}',
+                          'uniqueMember': memberdn}
+            usr_grp(multihost, group_info, 'group')
+        client.clear_sssd_cache()
+        ssh1 = client.auth_from_client(f'{user}', 'Secret123') == 3
+        assert ssh1
```

Comment:
Please add an assert message something along f"User {user} could not login via ssh."

Reply:
@jakub-vavra-cz the code is not ready for review yet. converting it to 'draft'. At the moment, ssh for users with different special characters is not working correctly.